internal/config/config.go

```diff
@@ -201,18 +208,66 @@ func getOutDir(outDirSpec, wd string) string {
 	return path.Join(wd, outDirSpec)
 }
 
-func defaultPackage(wd string) string {
-	for _, srcDir := range build.Default.SrcDirs() {
+func defaultPackage(wd string) (string, error) {
+	pkg, parent, err := currentModule()
+	if err != nil {
+		pkg = pkg + strings.TrimPrefix(wd, parent)
+		return pkg, nil
+	}
+
+	dirs := build.Default.SrcDirs()
+	for _, srcDir := range dirs {
 		if strings.HasPrefix(wd, srcDir) {
-			pkg := strings.Replace(wd, srcDir, "", -1)
-			return sanitizePackage(pkg)
+			pkg = strings.TrimPrefix(wd, srcDir)
+			pkg = sanitizePackage(pkg)
+			return pkg, nil
 		}
 	}
-	return sanitizePackage(wd)
+
+	return "", errors.New("can not determine default package name")
 }
 
 func sanitizePackage(pkg string) string {
 	pkg = filepath.ToSlash(pkg)
 	pkg = strings.TrimPrefix(pkg, "/")
 	return pkg
 }
+
+var errNoModules = errors.New("not using modules")
+
+// currentModule looks in all parent dirs for a go.mod file containing
+// a module name.
+func currentModule() (string, string, error) {
```

Comment:
O wow, great work.
This was all created before `go.mod` was a thing and since I have been out of the loop with developments in goworld, I totally missed this.
Thank you so much for adding this, this looks great :)